tutorials/Land/Heat/heat_equation.jl

```diff
@@ -0,0 +1,396 @@
+# - this method is only called at `t=0`
+# - `aux.z` and `aux.T` are available here because we've specified `z` and `T` in `vars_state_auxiliary`
+function init_state_auxiliary!(m::HeatModel, aux::Vars, geom::LocalGeometry)
+    aux.z = geom.coord[3]
+    aux.T = m.initialT
+end;
+
+# Specify the initial values in `state::Vars`. Note that
+# - this method is only called at `t=0`
+# - `state.ρcT` is available here because we've specified `ρcT` in `vars_state_conservative`
+function init_state_conservative!(
+    m::HeatModel,
+    state::Vars,
+    aux::Vars,
+    coords,
+    t::Real,
+)
+    state.ρcT = m.ρc * aux.T
+end;
+
+# The remaining methods, defined in this section, are called at every time-step in the solver by the [`BalanceLaw`](@ref ClimateMachine.DGMethods.BalanceLaw) framework.
+
+# Overload `update_aux!` to call `soil_nodal_update_aux!`, or any other auxiliary methods
+function update_aux!(
+    dg::DGModel,
+    m::HeatModel,
+    Q::MPIStateArray,
+    t::Real,
+    elems::UnitRange,
+)
+    nodal_update_aux!(soil_nodal_update_aux!, dg, m, Q, t, elems)
+    return true # TODO: remove return true
+end;
+
+# Compute/update all auxiliary variables at each node. Note that
+# - `aux.T` is available here because we've specified `T` in `vars_state_auxiliary`
+function soil_nodal_update_aux!(m::HeatModel, state::Vars, aux::Vars, t::Real)
+    aux.T = state.ρcT / m.ρc
+end;
+
+# Since we have diffusive fluxes, we must tell ClimateMachine to compute the gradient of `T`. Here, we specify how `T` is computed. Note that
+#  - `transform.T` is available here because we've specified `T` in `vars_state_gradient`
+function compute_gradient_argument!(
+    m::HeatModel,
+    transform::Vars,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    transform.T = state.ρcT / m.ρc
+end;
+
+# Specify where in `diffusive::Vars` to store the computed gradient in `compute_gradient_argument!`. Note that:
+#  - `diffusive.∇T` is available here because we've specified `∇T` in `vars_state_gradient_flux`
+#  - `∇transform.T` is available here because we've specified `T`  in `vars_state_gradient`
+function compute_gradient_flux!(
+    m::HeatModel,
+    diffusive::Vars,
+    ∇transform::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    diffusive.∇T = ∇transform.T
+end;
+
+# We do no have sources, nor non-diffusive fluxes.
+function source!(m::HeatModel, _...) end;
+function flux_first_order!(
+    m::HeatModel,
+    flux::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+) end;
+
+# Compute diffusive flux (``F(T,t) = -α ∇T`` in the original PDE). Note that:
+# - `diffusive.∇T` is available here because we've specified `∇T` in `vars_state_gradient_flux`
+function flux_second_order!(
+    m::HeatModel,
+    flux::Grad,
+    state::Vars,
+    diffusive::Vars,
+    hyperdiffusive::Vars,
+    aux::Vars,
+    t::Real,
+)
+    flux.ρcT -= m.α * diffusive.∇T
```

Comment:
I'd make the `gradient_flux` variable be `α∇T` as this ensures conservation of the diffusive heat flux and greatly simplifies applying Neumann boundary conditions (otherwise you technically have to multiply by `α⁻/α⁺` in L265 below).

Reply:
Thank you! I realized last night that I made similar mistakes in the initial docs. I'll ping you after adjusting to make sure that this part is correct!